src/compile/facet.ts

```diff
@@ -23,8 +23,13 @@ import {RepeaterValue, replaceRepeaterInFacet} from './repeater';
 import {parseGuideResolve} from './resolve';
 import {assembleDomain, getFieldFromDomain} from './scale/domain';
 
-export function facetSortFieldName(fieldDef: FacetFieldDef<string>, sort: EncodingSortField<string>, expr?: 'datum') {
-  return vgField(sort, {expr, suffix: `by_${vgField(fieldDef)}`});
+export function facetSortFieldName(
+  fieldDef: FacetFieldDef<string>,
+  sort: EncodingSortField<string>,
+  expr?: 'datum',
```

Comment:
Make `{expr, forAs=true}: {expr?: datum, forAs:boolean}`?

Reply:
No, we want the default to be false.

Reply:
Anyway, revised.